### DISCUSSION TITLE
Release v1.8.0

### DISCUSSION THREAD

**Discussion by tig** [tui-cs/Terminal.Gui#2030] [General]
Because this has new public APIs, we're bumping the verison to v1.8.

Current list of fixes:

- Fixes #2025 - API Docs are now generated via Github Action - View Source Works
- Fixes #1991 - Broken link in README
- Fixes #2026. Added ClearOnVisibleFalse to flag if the view must be cleared or not.
- Fixes #2017 and #2013. MainLoopTests.InvokeLeakTest failures
- Fixes #2014. Application mouseGrabView is run twice if return true.
- Fixes #2011. Wizard no longer needs to set specific colors, because #1971 has been fixed.
- Fixes #2006. ProgressBarStyles isn't disposing the _fractionTimer on quitting if running.
- Fixes #2004. TextFormatter.Justified not adding the extra spaces.
- Fixes #2002. Added feature to fill the remaining width with spaces.
- Fixes #1999. Prevents the mouseGrabView being executed with a null view.
- Fixes #1994. BREAKING CHANGE. Ensure only a single IdleHandlers list can exist.
- Fixes #1979. MessageBox.Query not wrapping since 1.7.1
- Fixes #1989. ListView: Ensures SelectedItem visibility on MoveDown and MoveUp.
- Fixes #1987. Textview insert text newline fix
- Fixes #1984. Setting Label.Visible to false does not hide the Label
- Fixes #820. Added HideDropdownListOnClick property.
- Fixes #1981. Added SplitNewLine method to the TextFormatter.
- Fixes #1973. Avoid positioning Submenus off screen.
- Added abstract MakeColor and CreateColors to create the colors at once.
- Fixes #1800. TextView now uses the same colors as TextField.
- Fixes #1969. ESC on CursesDriver take to long to being processed.
- Fixes #1967. New keys for DeleteAll on TextField and TextView.
- Fixes #1962 - Change KeyBindings to allow chaining commands
- Fixes #1961 Null reference in Keybindings Scenario and hotkey collision
- Fixes #1963. Only remove one character on backspace when wordwrap is on
- Fixes #1959. GoToEnd should not fail on an empty TreeView
- Fixes #1953. TextView cursor position is not updating by mouse.
- Fixes #1951. TextView with selected text doesn't scroll beyond the cursor position.
- Fixes #1948. Get unwrapped cursor position when word wrap is enabled on TextView.
- Ensures that the isButtonShift flag is disabled in all situations.
- Fixes #1943. Mouse ButtonShift is not preserving the text selected.

@BDisp is the updated `NStack` required for this?

**Comment by BDisp** [comment]
> @BDisp is the updated `NStack` required for this?

If NStack is updated then we can only use this statement:

`if (ustring.IsNullOrEmpty(text))`

otherwise we have to use the string keyword or:

`if (text != null && ustring.IsNullOrEmpty(text))`

But @tig the develop branch isn't correctly updated with the latest changes. You done something bad and is better take a better look of that.

**Reply by tig** [reply]
@BDisp I know it's screwed up. 

But I don't know how bad. Is it just #2023 that's broken, or did the other recent PRs get messed up too?

**Comment by tig** [comment]
PR #2031 - Release v1.8.0 is ready to be merged. Speak now or forever hold your peace...
